### PR TITLE
Filter auto

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [x] Chrome
- [ ] Firefox
- [x] Safari

## Checklist:

- [x] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [x] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/* tslint:disable */
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { SDSAutocomplteServiceInterface, SDSHiercarchicalServiceResult } from '@gsa-sam/components';
```
Comment (davereed) [inline libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts]: Spelling issue:
SDSAutocomplteServiceInterface

File: libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                items = result;
+            }
+        );
+        let totalItemCount = items.length;
+
+        let maxSectionPosition = currentItems + itemIncrease;
+        if (maxSectionPosition > totalItemCount) {
+            maxSectionPosition = totalItemCount;
+        }
+        let subItemsitems = items.slice(currentItems, maxSectionPosition);
```
Comment (davereed) [inline libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts]: Use a more readable variable name for subItemsitems

sectionItems maybe?

File: libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts [hunk 3]
```diff
@@ -1,144 +1,162 @@
...
+    this.settings3.primaryKeyField = 'id';
+    this.settings3.primaryTextField = 'name';
+    this.settings3.secondaryTextField = 'subtext';
+    this.settings3.labelText = 'Autocomplete 1';
+    this.settings3.selectionMode = SelectionMode.MULTIPLE;
+    this.settings3.autocompletePlaceHolderText = 'Enter text';
+    this.settings3.debounceTime = 0;
+
+
+    this.autocompleteModel3.items.push(this.data[0]);
```
Comment (davereed) [inline libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts]: Accessing arrays like this is generally unsafe. Prefer to use lodash/underscore .head method.

    function head(array) {
      return (array && array.length) ? array[0] : undefined;
    }

File: libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts [hunk 4]
```diff
@@ -1,144 +1,162 @@
...
+          modelChange: this.changes,
+         
+        },
       }]
     },
-
     {
-      key: 'testings',
+      key: 'filters',
       wrappers: ['accordianwrapper'],
```
Comment (davereed) [inline libs/sam-formly-sample/src/lib/feature/formly-sample/formly-input.component.ts]: Should be accordion not accordian

**Comment by davereed** [comment]
Check / improve coverage on:
sds-filters-comunication.service.ts
autocomplete-sample.service.ts

File: libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts (hunk 1 above)
Comment (yerramshilpa) [reply libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts]: This is the component service by Thomas. Can we discuss in the parling lot.

File: libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts (hunk 2 above)
Comment (yerramshilpa) [reply libs/sam-formly-sample/src/lib/feature/formly-sample/services/autocomplete-sample.service.ts]: updated..

**Comment by thomastighe** [comment]
Also why are you updating the CSS when there is another story for that?

**Comment by yerramshilpa** [comment]
> Also why are you updating the CSS when there is another story for that?

@thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.

**Comment by thomastighe** [comment]
> 
> 
> > Also why are you updating the CSS when there is another story for that?
> 
> @thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.

@yerramshilpa  there is an entire story to do the styles IAE-21642 holistically, Which is also assigned to Carlos this sprint.

File: libs/packages/components/src/lib/selected-result/selected-result.component.spec.ts [hunk 5]
```diff
@@ -5,6 +5,10 @@ import { SDSSelectedItemModel } from './models/sds-selectedItem.model';
 import { SDSSelectedItemModelHelper, SelectionMode } from './models/sds-selected-item-model-helper';
 import { By } from '@angular/platform-browser';
 import { SDSSelectedResultConfiguration } from './models/SDSSelectedResultConfiguration';
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { RouterModule } from '@angular/router';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/selected-result/selected-result.component.spec.ts]: removed

File: libs/packages/components/src/lib/selected-result/selected-result.component.html [hunk 6]
```diff
@@ -1,19 +1,21 @@
-<ul *ngIf="model" [attr.aria-label]="configuration.labelText +' results'" class="usa-unstyled-list resultsList"
+<ul *ngIf="model" [attr.aria-label]="configuration.labelText +' results'" class="usa-list usa-list--unstyled"
   aria-relevant="additions" role="listbox" aria-live="polite">
   <li role="option" *ngFor="let result of model.items; let i = index">
-    <div [attr.class]="disabled?'selected-item selected-item-disabled' :'selected-item'">
-      <ng-container *ngIf="itemTemplate" [ngTemplateOutlet]="itemTemplate"
-        [ngTemplateOutletContext]="{$implicit:result}">
+    <div>
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/selected-result/selected-result.component.html]: removed due to style changes, no need to disable the div, instead we can disable span (CHIP and Button).

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { Component, Input, ViewChild, TemplateRef, ElementRef, forwardRef } from '@angular/core';
+import { Component, Input, ViewChild, TemplateRef, ElementRef, forwardRef, OnInit } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.ts]: removed.

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.html [hunk 8]
```diff
@@ -1,8 +1,8 @@
 <div sds-click-outside (clickOutside)="checkForFocus($event)" sds-tab-outside (tabOutside)="checkForFocus($event)">
   <div class="input-container">
-    <input [disabled]="disabled" (ngModelChange)="textChange($event)" class="sam input" #input [attr.id]="configuration.id" autocomplete="off" type="text" (focus)="inputFocusHandler()"
-      (keydown)="onKeydown($event)" [(ngModel)]="inputValue" autocomplete="off" role="combobox" aria-autocomplete="list"
-      aria-expanded="false" aria-haspopup="true" [attr.aria-owns]="configuration.id+ '-listbox'" [attr.placeholder]="configuration.autocompletePlaceHolderText">
+    <input [disabled]="disabled" (ngModelChange)="textChange($event)"  class="usa-input margin-bottom-2" #input [attr.id]="configuration.id" autocomplete="off" type="text" (focus)="inputFocusHandler()"
+      (keyup)="onKeydown($event)" [(ngModel)]="inputValue" autocomplete="off" role="combobox" aria-autocomplete="list"
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.html]: before you made changes, I was working on the file it was a merge conflict issue.

File: libs/packages/components/src/lib/public-api.ts [hunk 9]
```diff
@@ -4,7 +4,7 @@ export * from './toolbar/toolbar.module';
 export * from './header/header.module';
 export * from './footer/footer.module';
 export * from './pagination/pagination.module';
-
+export * from './autocomplete/autocomplete.component';
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/public-api.ts]: I was using the individual component in viewchild, if we don't export it was not working.

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts [hunk 10]
```diff
@@ -67,21 +67,18 @@ describe('SamAutocompleteComponent', () => {
...
   }));
 
   it('Should have results with minimumCharacterCountSearch', fakeAsync(() => {
-
-    const event ='Level 7' 
-    component.configuration.minimumCharacterCountSearch = 3;
+    
+    const event ='Level 7';
     component.textChange(event);
+    component.configuration.minimumCharacterCountSearch = 3;
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts]: reverted changes

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.html (hunk 8 above)
[the remark this answers is not in the document]
Comment (thomastighe) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.html]: Can you please change this back to keydown

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts [hunk 11]
```diff
@@ -67,21 +67,18 @@ describe('SamAutocompleteComponent', () => {
     fixture.detectChanges();
     const list = fixture.debugElement.query(By.css('.autocomplete-result'));
     expect(list.nativeElement.children.length).toBe(1);
-    const emptyItem = fixture.debugElement.query(By.css('.emptyResults'));
```
[the remark this answers is not in the document]
Comment (thomastighe) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts]: You did not address this being removed

File: libs/packages/components/src/lib/public-api.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (thomastighe) [reply libs/packages/components/src/lib/public-api.ts]: @davereed  Am I missing something if you import the module into the sam formly module this should not need to be done. I could be mistaken on this but I do not think this should be done.

File: libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/autocomplete-search/autocomplete-search.component.spec.ts]: I already reverted those changes

File: libs/packages/components/src/lib/public-api.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (davereed) [reply libs/packages/components/src/lib/public-api.ts]: Looks like she tried it the other way and it didn't work. Let's revisit this in the IP sprint to see if we can avoid importing the component and prefer to import the module only.

**Comment by yerramshilpa** [comment]
> > > Also why are you updating the CSS when there is another story for that?
> > 
> > 
> > @thomastighe I thought you already know what I have updated it, but it seems you didn't get what I have updated it. I have not added any new CSS, I have updated with pre-existing styles which you should have done it earlier. There are other CSS works need to be done, those can be done on the ticket.
> 
> @yerramshilpa there is an entire story to do the styles IAE-21642 holistically, Which is also assigned to Carlos this sprint.

@thomastighe Carlos updated the styles

**Pull request merged**